reporter/src/funTest/kotlin/reporters/CycloneDxReporterFunTest.kt

```diff
@@ -23,41 +23,151 @@ package org.ossreviewtoolkit.reporter.reporters
+import io.kotest.matchers.collections.shouldHaveSize
 import io.kotest.matchers.file.aFile
 import io.kotest.matchers.file.emptyFile
 import io.kotest.matchers.should
 import io.kotest.matchers.shouldBe
 import io.kotest.matchers.shouldNotBe
 
+import java.io.File
+
 import org.cyclonedx.parsers.JsonParser
 import org.cyclonedx.parsers.XmlParser
 
 import org.ossreviewtoolkit.reporter.ORT_RESULT
 import org.ossreviewtoolkit.reporter.ReporterInput
+import org.ossreviewtoolkit.reporter.patchCycloneDxResult
 import org.ossreviewtoolkit.utils.test.createSpecTempDir
 
 class CycloneDxReporterFunTest : WordSpec({
     val defaultSchemaVersion = CycloneDxReporter.DEFAULT_SCHEMA_VERSION.versionString
-    val options = mapOf("single.bom" to "true")
+    val optionSingle = mapOf("single.bom" to "true")
+    val optionMulti = mapOf("single.bom" to "false")
     val outputDir = createSpecTempDir()
 
-    "A generated BOM" should {
+    "BOM generation with single option" should {
+        "be composed with just one file" {
+            val jsonOptions = optionSingle + mapOf("output.file.formats" to "json")
+            val bomFiles = CycloneDxReporter().generateReport(ReporterInput(ORT_RESULT), outputDir, jsonOptions)
+
+            bomFiles shouldHaveSize 1
+        }
+
         "be valid XML according to schema version $defaultSchemaVersion" {
-            val xmlOptions = options + mapOf("output.file.formats" to "xml")
+            val xmlOptions = optionSingle + mapOf("output.file.formats" to "xml")
             val bomFile = CycloneDxReporter().generateReport(ReporterInput(ORT_RESULT), outputDir, xmlOptions).single()
 
             bomFile shouldBe aFile()
             bomFile shouldNotBe emptyFile()
             XmlParser().validate(bomFile, CycloneDxReporter.DEFAULT_SCHEMA_VERSION) should beEmpty()
         }
 
+        "create the expected XML file" {
+            val expectedBom = File("src/funTest/assets/bom.cyclonedx.xml").readText()
+
+            val xmlOptions = optionSingle + mapOf("output.file.formats" to "xml")
+            val bomFile = CycloneDxReporter().generateReport(ReporterInput(ORT_RESULT), outputDir, xmlOptions).single()
+
+            bomFile.readText().patchCycloneDxResult() shouldBe expectedBom
+        }
+
         "be valid JSON according to schema version $defaultSchemaVersion" {
-            val jsonOptions = options + mapOf("output.file.formats" to "json")
+            val jsonOptions = optionSingle + mapOf("output.file.formats" to "json")
             val bomFile = CycloneDxReporter().generateReport(ReporterInput(ORT_RESULT), outputDir, jsonOptions).single()
 
             bomFile shouldBe aFile()
             bomFile shouldNotBe emptyFile()
             JsonParser().validate(bomFile, CycloneDxReporter.DEFAULT_SCHEMA_VERSION) should beEmpty()
         }
+
+        "create the expected JSON file" {
+            val expectedBom = File("src/funTest/assets/bom.cyclonedx.json").readText().replace("\n", "")
```

Comment:
The `replace("\n", "")` calls should not be necessary / avoided. If line ending normalization is required, instead ensure that you add a line similar to the existing line `scanner/src/test/assets/*.json  text eol=lf` to `.gitattributes`.